firefox_puppeteer/ui/toolbars.py

```diff
@@ -0,0 +1,337 @@
+
+    def reload_url(self, evt='shortcut', force=False):
+        """Reload the currently open page.
+
+        :param evt: The event type to use to cause the reload. (one of
+                    "shortcut", "shortcut2", or "button").
+        :param force: Whether to cause a forced reload.
+        """
+        # TODO: The force parameter is ignored for the moment. Use
+        # mouse event modifiers or actions when they're ready.
+        # Bug 1097705 tracks this feature in marionette.
+        if evt == 'button':
+            self.reload_button.click()
+        elif evt == 'shortcut':
+            cmd_key = self.l10n.get_localized_entity(LocationBar.dtds,
+                                                     'reloadCmd.commandkey')
+            self.urlbar.send_keys(cmd_key)
+        elif evt == 'shortcut2':
+            self.urlbar.send_keys(self.keys.F5)
+
+    def focus(self, evt='click'):
+        """Focus the location bar according to the provided event.
+
+        :param evt: The event to synthesize in order to focus the urlbar
+                    (one of 'click' or 'shortcut').
+        """
+        if evt == 'click':
+            self.urlbar.click()
+        elif evt == 'shortcut':
+            cmd_key = self.l10n.get_localized_entity(LocationBar.dtds,
+                                                     'openCmd.commandkey')
+            (self.marionette.find_element(By.ID, 'main-window')
+                            .send_keys(self.keys.ACCEL, cmd_key))
+        else:
+            raise ValueError("An unknown event type was passed: %s" % evt)
+
+        Wait(self.marionette).until(
+            lambda _: self.urlbar.get_attribute('focused') == 'true')
+
+
+class AutocompleteResults(BaseLib):
+    """Library for interacting with autocomplete results.
+    """
+
+    def __init__(self, *args, **kwargs):
+        BaseLib.__init__(self, *args, **kwargs)
+        # TODO: A "utility" module that sets up the client directly would be
+        # useful here.
+        self.l10n = L10n(self.get_marionette)
+        self.keys = Keys(self.get_marionette)
+
+    @property
+    def popup(self):
+        """Returns a list of auto complete result elements.
+        """
+        return self.marionette.find_element(By.ID,
+                                            'PopupAutoCompleteRichResult')
+
+    @property
+    def results(self):
+        """Returns the container node for autocomplete results.
+        """
+        return self.popup.find_element('anon attribute',
+                                       {'anonid': 'richlistbox'})
+
+    def is_open(self):
+        return self.popup.get_attribute('state') == 'open'
+
+    def get_visible_results(self):
+        """Returns a list of visible result nodes."""
+        return self.marionette.execute_script("""
+          let rv = [];
+          let node = arguments[0];
+          let count = node.itemCount;
+          for (let i = 0; i < count; ++i) {
```

Comment:
nit: not really a need to define count here, just put node.itemCount into the brackets directly.